nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -141,3 +139,64 @@ def __exit__(self, *args):
 
     def __del__(self):
         self.vc.release()
+
+
+class MovieDataChunkIterator(GenericDataChunkIterator):
+    """DataChunkIterator specifically for use on RecordingExtractor objects."""
+
+    def __init__(
+        self,
+        movie_file: FilePathType,
+        buffer_gb: float = None,
+        chunk_shape: tuple = None,
+        stub_test: bool = False,
+    ):
+        self.video_capture_ob = VideoCaptureContext(movie_file)
+        self._full_frame_size_mb, self._full_frame_shape = self._get_frame_details()
+        if stub_test:
+            self.video_capture_ob.frame_count = 10
+        super().__init__(
+            buffer_gb=buffer_gb,
+            chunk_shape=chunk_shape,
+            display_progress=True,
+        )
+
+    def _get_default_chunk_shape(self, chunk_mb):
+        """Shape is either one frame or a subset: scaled frame size but with all pixel colors"""
+        num_frames = int(chunk_mb // self._full_frame_size_mb)
+        num_frames = 1 if num_frames is 0 else num_frames
```

Comment:
```suggestion
        num_frames = int(np.ceil(chunk_mb / self._full_frame_size_mb))
```
Would this work?

Reply:
this won't because I want `np.ceil` only if the value is between 0 and 1. Else I want `np.floor`.

Reply:
👍